### ISSUE TITLE
DismisAll doesn't work if multiple toasts called at the same time

### ISSUE THREAD

**Issue by amxDusT** [payam-zahedi/toastification#119]
**Describe the bug**
I'm trying to show only one toast at a time in my application by dismissing all before showing new toasts, but whenever 2 or more toasts get called at around the same time, they all show instead of dismissing.

**To Reproduce**
```dart
toastification
    ..dismissAll(delayForAnimation: false)
    ..show(
      title: const Text('test 1'),
    )
    ..dismissAll(delayForAnimation: false)
    ..show(
      title: const Text('test 2'),
    );
``` 

**Expected behavior**
The first toast ('test 1') should be dismissed

**Flutter information:**
 - OS: iOS, Android
 - Flutter Version: 3.22.3
 - Toastification Version 2.1.0

**Comment by payam-zahedi** [comment]
Hey @amxDusT 

Thanks for opening this issue

what exactly you want to achive here?

To only show one notification?

**Comment by amxDusT** [comment]
Exactly.
The problem seems to be in the toastification_manager: it has a delay in showing the toast, apparently waiting for the overlay. Though when dismiss/dismissAll is called these delays don't get removed.
The following pull is an attempt to fix that #123

**Comment by payam-zahedi** [comment]
Alright, I got your point

but I'm not sure the best way to solve it is to keep some toast as idle and keep them in a separate list 

let me investigate on this and will let you know what's my plan.
